### PR TITLE
sql: introduce datafuse standard AST and transformer from `sqlparser-rs`

### PR DESCRIPTION
**Pull request by leiysky**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

This PR introduce a set of SQL AST, and transformer from AST produced by `sqlparser-rs`.

## Changelog

- Improvement

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: query/src/sql/parser/mod.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            .into_iter()
+            .map(|stmt| {
+                let transformer = AstTransformerFactory::new_sqlparser_transformer(stmt);
+                transformer.transform()
+            })
+            .collect::<Result<_>>()
+    }
+}
+
+#[cfg(test)]
```
Comment (BohuTANG) [inline query/src/sql/parser/mod.rs:47]: Cool, it's better to move the test to a spereate file like x_test.rs

Comment (leiysky) [reply query/src/sql/parser/mod.rs:47]: Yes, I'm working on this.

Comment (leiysky) [reply query/src/sql/parser/mod.rs]: Done.

**Comment by leiysky** [comment]
I'm going to add more tests.

File: query/src/sql/parser/transformer/transform_sqlparser.rs [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+        match data_type {
+            SqlparserDataType::Char(length) => Ok(TypeName::Char(length.to_owned())),
+            SqlparserDataType::Varchar(length) => Ok(TypeName::Varchar(length.to_owned())),
+            SqlparserDataType::Decimal(prec, scale) => {
+                Ok(TypeName::Decimal(prec.to_owned(), scale.to_owned()))
+            }
+            SqlparserDataType::Float(length) => Ok(TypeName::Float(length.to_owned())),
+            SqlparserDataType::Int => Ok(TypeName::Int),
+            SqlparserDataType::BigInt => Ok(TypeName::BigInt),
+            SqlparserDataType::Real => Ok(TypeName::Real),
```
Comment (sundy-li) [inline query/src/sql/parser/transformer/transform_sqlparser.rs:571]: SmallInt and TinyInt

**Comment by sundy-li** [comment]
Maybe this pr can introduce `params` in `Expr::Function`?

https://github.com/datafuse-extras/sqlparser-rs/pull/5

File: query/src/sql/parser/transformer/transform_sqlparser.rs (hunk 2 above)
Comment (leiysky) [reply query/src/sql/parser/transformer/transform_sqlparser.rs:575]: Done

**Comment by leiysky** [comment]
> Maybe this pr can introduce `params` in `Expr::Function`?
> 
> [datafuse-extras/sqlparser-rs#5](https://github.com/datafuse-extras/sqlparser-rs/pull/5)

I've just supported it in this PR, but the transformer part is blocked by bumping `sqlparser-rs` version.
https://github.com/datafuselabs/datafuse/blob/25aca3c35c1598b4a2dbc02d5e6419b6378361da/query/src/sql/parser/transformer/transform_sqlparser.rs#L482-L483

I think we can fix this later.

**Review by sundy-li — APPROVED** [review]
LGTM

**Pull request merged**